base/compiler/ssair/passes.jl

```diff
@@ -1075,31 +1075,26 @@ function try_inline_finalizer!(ir::IRCode, argexprs::Vector{Any}, idx::Int, mi::
     # Ok, we're committed to inlining the finalizer
     et !== nothing && push!(et, mi)
 
-    linetable_offset, extra_coverage_line = ir_inline_linetable!(ir.linetable, src, mi.def, ir[SSAValue(idx)][:line])
-    if extra_coverage_line != 0
-        insert_node!(ir, idx, NewInstruction(Expr(:code_coverage_effect), Nothing, extra_coverage_line))
-    end
+    # TOOD: Should there be a special line number node for inlined finalizers?
+    insert_node_here! = new_inst->insert_node!(ir, idx, new_inst)
+    inlined_at = ir[SSAValue(idx)][:line]
```

Comment:
Possibly it should be `inlined_at = 0`? I think codegen will interpret that as being inlined at the topmost line (if it needs to switch files) or just take it explicitly if it is in the same file. Or is the idea that it should appear to preserve the context information of the point where we did the inlining at?

Either way, this is only supposed to be visible in a debugger (since it is nothrow), so I guess it matters relatively little.

Reply:
Yeah, it's not clear what it should be. I think this is a reasonable choice, because it inlines at the point that the code is actually inserted, which I think will produce the most useful backtraces, but I think in general, we may want to annotate such frames in the backtrace.